### PR TITLE
Polish graph

### PR DESCRIPTION
**Pull request by daming-lu**
add slider
add restore image
add panning and zooming to graph

### CODE REVIEW

File: frontend/src/graph/ui/Chart.vue [hunk 1]
```diff
@@ -27,22 +28,74 @@ export default {
...
+      this.restoreImage(false);
+    },
+    scale: function(val) {
+      let k = this.zoomScale(val);
+      let svg = d3.select('svg');
+      svg.call(this.graphZoom.transform, d3.zoomIdentity.scale(k));
+    },
   },
+  methods: {
+    restoreImage(thenDownload) {
```
Comment (jetfuel) [inline frontend/src/graph/ui/Chart.vue:72]: I feel we should divide this into two functions. 
restore Image and download images are not tightly related. But I understand if there is technical difficulties, we can leave them as it is for now. 

Let's leave some more comments to explain this edge case. Should be helpful later.

File: frontend/src/graph/ui/Chart.vue [hunk 2]
```diff
@@ -27,22 +28,74 @@ export default {
...
+      let chartScope = this;
+      let svg = d3.select('svg');
+      this.imageWidth = this.originImageWidth;
+      this.imageHeight = this.originImageHeight;
 
+      if (thenDownload) {
+        svg.transition()
```
Comment (jetfuel) [inline frontend/src/graph/ui/Chart.vue:79]: Ah, I see. We can use this callback function to download the images. Nice.

**Review by jetfuel — DISMISSED** [review]
LGTM

File: .travis.yml [hunk 3]
```diff
@@ -52,7 +52,7 @@ before_install:
 
 script:
   - if [[ "$JOB" == "check_style" ]]; then ./scripts/check_style.sh; fi
-  - if [[ "$JOB" == "test" ]]; then ./scripts/tests.sh all; WITH_PYTHON3=ON ./scripts/tests.sh all; fi
```
Comment (daming-lu) [inline .travis.yml:55]: Disable Py3 test for now as Travis CI is having problem installing 'six' package for Py2Py3 compatibility.

**Review by jetfuel — APPROVED** [review]

**Pull request merged**